src/Analyzers/CSharp/CodeFixes/UseCollectionInitializer/CSharpUseCollectionInitializerCodeFixProvider_CollectionExpression.cs

```diff
@@ -2,588 +2,38 @@
-
-                return indentationString;
-            }
-
-            bool MakeMultiLineCollectionExpression()
-            {
-                var totalLength = 0;
-                if (objectCreation.Initializer != null)
-                {
-                    foreach (var expression in objectCreation.Initializer.Expressions)
-                        totalLength += expression.Span.Length;
-                }
-
-                foreach (var (statement, _) in matches)
-                {
-                    // if the statement we're replacing has any comments on it, then we need to be multiline to give them an
-                    // appropriate place to go.
-                    if (statement.GetLeadingTrivia().Any(static t => t.IsSingleOrMultiLineComment()) ||
-                        statement.GetTrailingTrivia().Any(static t => t.IsSingleOrMultiLineComment()))
-                    {
-                        return true;
-                    }
-
-                    foreach (var component in GetElementComponents(statement))
-                    {
-                        // if any of the expressions we're adding are multiline, then make things multiline.
-                        if (!document.Text.AreOnSameLine(component.GetFirstToken(), component.GetLastToken()))
-                            return true;
-
-                        totalLength += component.Span.Length;
-                    }
-                }
-
-                return totalLength > wrappingLength;
-            }
-
-            static IEnumerable<SyntaxNode> GetElementComponents(StatementSyntax statement)
-            {
-                if (statement is ExpressionStatementSyntax expressionStatement)
-                {
-                    yield return expressionStatement.Expression;
-                }
-                else if (statement is ForEachStatementSyntax foreachStatement)
-                {
-                    yield return foreachStatement.Expression;
-                }
-                else if (statement is IfStatementSyntax ifStatement)
-                {
-                    yield return ifStatement.Condition;
-                    yield return UnwrapEmbeddedStatement(ifStatement.Statement);
-                    if (ifStatement.Else != null)
-                        yield return UnwrapEmbeddedStatement(ifStatement.Else.Statement);
-                }
-            }
-
-            static StatementSyntax UnwrapEmbeddedStatement(StatementSyntax statement)
-                => statement is BlockSyntax { Statements: [var innerStatement] } ? innerStatement : statement;
+            return CSharpCollectionExpressionRewriter.CreateCollectionExpressionAsync(
+                document,
+                fallbackOptions,
+                objectCreation,
+                matches.SelectAsArray(m => new CollectionExpressionMatch(m.Statement, m.UseSpread)),
+                o => o.Initializer,
+                (o, i) => o.WithInitializer(i),
+                cancellationToken);
```

Comment:
exactly the same as before, except:

1. Converting this feature's 'Match' type to 'CollectionExpressionMatch'
2. Passing the lambdas to make this code work with ObjectCreationExpressions